src/treetensornetwork/projttnosum.jl

```diff
@@ -0,0 +1,61 @@
+"""
+ProjTTNOSum
+"""
+mutable struct ProjTTNOSum
```

Comment:
Again, I think this should not be `mutable`.

I was also thinking about using the `Applied` system in `ITensors.LazyApply` to avoid having to actually define types like this. That would be a more general way to represent lazy combinations of projected TTNs (i.e. help with representing more general combinations, like sums of projected TTNS and TTNO, sums of TTNOs applied to TTNS, etc. without having to hardcode specific types).

Maybe we could chat about how that might work, this would be a good opportunity to rethink some of the type designs here (some of these designs are holdovers from porting the C++ version).

Reply:
This definitely sounds like a nice idea.